lib/blacklight_advanced_search/controller.rb

```diff
@@ -14,44 +14,35 @@ module BlacklightAdvancedSearch::Controller
     self.blacklight_config.advanced_search[:form_solr_parameters] ||= {}    
     
     
-    # Parse app URL params used for adv searches 
-    self.solr_search_params_logic += [:add_advanced_search_to_solr]
-    
+    if self.respond_to? :search_params_logic
+      # Parse app URL params used for adv searches 
+      self.search_params_logic += [:add_advanced_search_to_solr]
+    end
+
+    if self.blacklight_config.search_builder_class
+    self.blacklight_config.search_builder_class.send(:include,  
```

Comment:
Yeah, this is not great. I guess we should have a Blacklight::SearchBuilderBehavior module in Blacklight so that the generated class only has to be 3 lines.